Source/SwiftLintFramework/Rules/ExplicitAssociatedEnumValueRule.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  ExplicitAssociatedEnumValueRule.swift
+//  SwiftLint
+//
+//  Created by Mazyad Alabduljaleel on 8/19/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ExplicitAssociatedEnumValueRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_associated_enum_value",
+        name: "Explicit Associated Enum Value",
+        description: "Enums should be explicitly assigned their associated values.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            "enum Numbers {\n case int(Int)\n case short(Int16)\n}\n",
+            "enum Numbers: Int {\n case one = 1\n case two = 2\n}\n",
+            "enum Numbers: Double {\n case one = 1.1\n case two = 2.2\n}\n",
+            "enum Numbers: String {\n case one = \"one\"\n case two = \"two\"\n}\n",
+            "enum Numbers: String {\n case one = \"ONE\"\n case two = \"TWO\"\n}\n"
+        ],
+        triggeringExamples: [
+            "enum Numbers: Int {\n case one = 10, ↓two, three = 30\n}\n",
+            "enum Numbers: String {\n case ↓one\n case ↓two\n}\n",
+            "enum Numbers: String {\n case ↓one, two = \"two\"\n}\n",
+            "enum Numbers: String {\n case ↓one, ↓two\n}\n"
+        ]
+    )
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .enum else {
+            return []
+        }
+
+        // Check if it's an associated value enum
+        guard !dictionary.inheritedTypes.isEmpty else {
```

Comment:
this is not enough, because enums can implement protocols, for example:

```swift
protocol Foo {}
enum Bar: Foo {
  case bar
}
```

Instead, we should keep a list of possible raw types that support implicit raw values. As far as I know, that means `String` and all numeric types (`Int`, `Float`, `Int64`, ...). Types like `Character` conform to `RawRepresentable`, but they need an explicit value.